src/appleseedmaya/renderviewtilecallback.cpp

```diff
@@ -115,14 +116,19 @@ namespace
             write_tile(frame, tile_x, tile_y);
         }
 
-        void on_progressive_frame_update(const asr::Frame* frame) override
+        void on_progressive_frame_update(
+            const asr::Frame&       frame,
+            const double            time,
+            const std::uint64_t     samples,
+            const double            samples_per_pixel,
+            const std::uint64_t     samples_per_second) override
         {
-            const asf::CanvasProperties& props = frame->image().properties();
+            const asf::CanvasProperties& props = frame.image().properties();
 
             for (size_t ty = 0; ty < props.m_tile_count_y; ++ty)
             {
                 for (size_t tx = 0; tx < props.m_tile_count_x; ++tx)
-                    write_tile(frame, tx, ty);
+                    write_tile(&frame, tx, ty);
```

Comment:
Ideally the opposite change would be done: `write_tile()`'s first argument should be a reference, not a pointer (because it must not be null).

Reply:
True, but the same function is also used inside the overridden method on_tile_end() which expects a pointer, not reference. If I change write_tile() to accept a reference as argument then I get some issues. I will add an assert in write_tile() to check that Frame* is not a null-pointer.

Reply:
> True, but the same function is also used inside the overridden method on_tile_end() which expects a pointer, not reference.

Yes but then you can simply pass `*frame` instead of `frame`.

Anyway, not a big deal.